src/main/java/io/confluent/kafka/connect/datagen/DatagenTask.java

```diff
@@ -182,6 +182,12 @@ public void start(Map<String, String> props) {
     }
 
     avroSchema = generator.schema();
+
+    if (!schemaKeyField.isEmpty() && avroSchema.getField(schemaKeyField) == null) {
+      throw new ConnectException("Schema does not contain the field '" + schemaKeyField
+              + "' provided in config '" + DatagenConnectorConfig.SCHEMA_KEYFIELD_CONF + "'");
```

Comment:
The [ConfigException](https://kafka.apache.org/28/javadoc/org/apache/kafka/common/config/ConfigException.html) class seems perfect for this case:
```suggestion
      throw new ConfigException(schemaKeyField, DatagenConnectorConfig.SCHEMA_KEYFIELD_CONF, "Schema does not contain the specified field");
```

(If the suggestion is applied, it'll also be necessary to add an import to the top of this file and to change the expected exception type in the unit test)

Reply:
Thanks, that's perfect!